packages/next/build/webpack-config.ts

```diff
@@ -192,25 +64,125 @@ export default function getBaseWebpackConfig (dir, { dev = false, isServer = fal
 
   const webpackMode = dev ? 'development' : 'production'
 
-  let webpackConfig = {
+  const terserPluginConfig = {
+    parallel: true,
+    sourceMap: false,
+    cache: true,
+    cacheKeys: (keys: any) => {
+      // path changes per build because we add buildId
+      // because the input is already hashed the path is not needed
+      delete keys.path
+      return keys
+    }
+  }
+
+  let webpackConfig: webpack.Configuration = {
     mode: webpackMode,
     devtool: dev ? 'cheap-module-source-map' : false,
     name: isServer ? 'server' : 'client',
     target: isServer ? 'node' : 'web',
-    externals: externalsConfig(isServer, target),
-    optimization: optimizationConfig({ dir, dev, isServer, totalPages, target }),
+    externals: isServer && target !== 'serverless' ? [
+      (context, request, callback) => {
+        const notExternalModules = ['next/app', 'next/document', 'next/link', 'next/router', 'next/error', 'http-status', 'string-hash', 'ansi-html', 'hoist-non-react-statics', 'htmlescape']
+
+        if (notExternalModules.indexOf(request) !== -1) {
+          return callback()
+        }
+    
+        resolve(request, { basedir: context, preserveSymlinks: true }, (err, res) => {
+          if (err) {
+            return callback() //TODO: We don't return the error?
```

Comment:
```suggestion
            return callback()
```

This is expected as we just mark the file as "let webpack bundle it"